extensions/iceberg/src/main/java/io/deephaven/iceberg/location/IcebergTableParquetLocationKey.java

```diff
@@ -51,4 +111,117 @@ public String getImplementationName() {
     public ParquetInstructions readInstructions() {
         return readInstructions;
     }
+
+    /**
+     * When comparing with another {@link IcebergTableParquetLocationKey}, precedence-wise this implementation compares:
+     * <ul>
+     * <li>{@code order}</li>
+     * <li>{@code uuid}, if not null, else:
+     * <ul>
+     * <li>{@code catalogName}</li>
+     * <li>{@code tableIdentifier}</li>
+     * </ul>
+     * </li>
+     * <li>{@code dataSequenceNumber}</li>
+     * <li>{@code fileSequenceNumber}</li>
+     * <li>{@code manifestSequenceNumber}</li>
+     * <li>{@code dataFilePos}</li>
+     * <li>{@code uri}</li>
+     * </ul>
+     * Otherwise, it delegates to the parent class.
+     * <p>
+     * {@inheritDoc}
+     */
+    @Override
+    public int compareTo(@NotNull final TableLocationKey other) {
+        if (other instanceof IcebergTableParquetLocationKey) {
+            // TODO(deephaven-core#5989): Add unit tests for the ordering of data files
+            final IcebergTableParquetLocationKey otherTyped = (IcebergTableParquetLocationKey) other;
+            int comparisonResult;
+            if ((comparisonResult = Integer.compare(order, otherTyped.order)) != 0) {
+                return comparisonResult;
+            }
+            // Establish an ordering for data files from different tables
+            if (tableUuid != null && otherTyped.tableUuid != null) {
+                if ((comparisonResult = tableUuid.compareTo(otherTyped.tableUuid)) != 0) {
+                    return comparisonResult;
+                }
+            } else {
+                if ((comparisonResult = catalogName.compareTo(otherTyped.catalogName)) != 0) {
+                    return comparisonResult;
+                }
```

Comment:
That's precisely why I think it should be included. If the user were to create a TLP sourcing from two distinctly named catalogs, I'd want them to be able to include an otherwise-identical location from each. Maybe it's too hypothetical.